### PR TITLE
NO-SNOW Remove unused code path from RecordService

### PR DESCRIPTION
**Pull request by sfc-gh-mbobowski**
# Overview

NO-SNOW Remove unused code path from RecordService

RecordService class parameter `behaviorOnNullValues` is always set to DEFAULT and is not modified in the production code. 

## Pre-review checklist
- [ ] This change should be part of a Behavior Change Release. See [go/behavior-change](http://go/behavior-change).
- [ ] This change has passed Merge gate tests
- [ ] Snowpipe Changes
- [x] Snowpipe Streaming Changes
- [ ] This change is TEST-ONLY
- [ ] This change is README/Javadocs only
- [ ] This change is protected by a config parameter <PARAMETER_NAME> eg `snowflake.ingestion.method`.
    - [ ] `Yes` - Added end to end and Unit Tests. 
    - [ ] `No` - Suggest why it is not param protected
- [ ] Is his change protected by parameter <PARAMETER_NAME> on the server side?
    - [ ] The parameter/feature is not yet active in production (partial rollout or PrPr, see [Changes for Unreleased Features and Fixes](http://go/ppp-prpr)).
    - [ ] If there is an issue, it can be safely mitigated by turning the parameter off. This is also verified by a test (See [go/ppp](http://go/ppp)).

### CODE REVIEW

**Review by sfc-gh-japatel — APPROVED** [review]
I understand this is not used in prod code but I guess @sfc-gh-rcheng would have some reason to test this path. 
Please wait for her to approve it too, but lgtm!

**Review by sfc-gh-tzhang — APPROVED** [review]
lgtm, thanks

**Review by sfc-gh-lshcharbaty — APPROVED** [review]

**Review by sfc-gh-rcheng — APPROVED** [review]
lgtm thanks!

**Pull request merged**
